src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowRun.java

```diff
@@ -1277,6 +1110,15 @@ public void save() throws IOException {
         synchronized (this) {
             PipelineIOUtils.writeByXStream(this, loc, XSTREAM2, isAtomic);
             SaveableListener.fireOnChange(this, file);
+            if (Boolean.TRUE.equals(completed)) {
+                Executor executor = getExecutor();
+                if (executor != null) {
+                    AsynchronousExecution asynchronousExecution = executor.getAsynchronousExecution();
+                    if (asynchronousExecution != null) {
+                        asynchronousExecution.completed(null);
+                    }
```

Comment:
Maybe log a warning if `asynchronousExecution`  is null? I don't think it should ever happen, but we've seen some bugs where OneOffExecutors stick around as zombies even when their associated Pipelines appear to have completed (e.g., [JENKINS-50199](https://issues.jenkins-ci.org/browse/JENKINS-50199), [JENKINS-45571](https://issues.jenkins-ci.org/browse/JENKINS-45571)), so it could end up being useful one day.

Reply:
Probably unrelated.